### ISSUE TITLE
Uuid query bind & result try_into invalid

### ISSUE THREAD

**Issue by searedcircuit** [stargate/stargate-grpc-rust-client#18]
Uuid conversion is not working in 0.3, see the following samples & errors:

> `let userid = uuid::Uuid::new_v4();
> let query = Query::builder().keyspace("...").query("...").bind((userid)).build();`
> 
> Error:
> 
> > the trait bound `uuid::Uuid: DefaultCqlType` is not satisfied
> > required because of the requirements on the impl of `From<uuid::Uuid>` for `Value`
> > required because of the requirements on the impl of `Into<Value>` for `uuid::Uuid`
> > required because of the requirements on the impl of `IntoValue<stargate_grpc::types::Any>` for `uuid::Uuid`
> > 2 redundant requirements hidden
> > required because of the requirements on the impl of `Into<stargate_grpc::proto::Values>` for `(uuid::Uuid, std::string::String, std::string::String, std::string::String, bool, i64, i64)`

> 
> `let userid: Uuid = userid_res.rows.first().try_into()?;`
> 
> Error:
> 
> > the trait bound `uuid::Uuid: From<std::option::Option<&Row>>` is not satisfied
> > required because of the requirements on the impl of `Into<uuid::Uuid>` for `std::option::Option<&Row>`
> > required because of the requirements on the impl of `TryFrom<std::option::Option<&Row>>` for `uuid::Uuid`
> > required because of the requirements on the impl of `TryInto<uuid::Uuid>` for `std::option::Option<&Row>`

**Comment by mlr** [comment]
@pkolaczk do you know what might be the cause? I was able to reproduce by trying to use (and slightly adapt to be precise) the uuid example code. I can share a gist if helpful.

**Comment by pkolaczk** [comment]
Can you show me your Cargo.toml and the command line you're building and running the program?
Feature `uuid` must be enabled explicitly to allow conversions to/from `uuid::Uuid`.

**Comment by searedcircuit** [comment]
```
uuid = { version = "0.8.2", features = ["serde","v4"] }
stargate-grpc = {version = "0.3.0", features = ["uuid","macros","chrono"]}
stargate-grpc-derive = "0.2.0"
```

I didn't notice this at first, and it did help with conversion _to_ cql, but the reverse still fails.

**Comment by pkolaczk** [comment]
ok, so after adding the "uuid" feature, can you paste the actual code that's causing the problem? 

```rust
let userid: Uuid = userid_res.rows.first().try_into()?;
```

That doesn't look right IMHO, because you're trying to convert the whole `Row` into a single `Uuid`. 
You probably wanted to convert a column value to a `Uuid`.
Have you tried:

```rust
let userid: Uuid = userid_res.rows.first().try_take(0)?;
```

**Comment by searedcircuit** [comment]
> ok, so after adding the "uuid" feature, can you paste the actual code that's causing the problem?
> 
> ```rust
> let userid: Uuid = userid_res.rows.first().try_into()?;
> ```
> 
> That doesn't look right IMHO, because you're trying to convert the whole `Row` into a single `Uuid`. You probably wanted to convert a column value to a `Uuid`. Have you tried:
> 
> ```rust
> let userid: Uuid = userid_res.rows.first().try_take(0)?;
> ```

rows.first() returns option, so I think it would have to be something closer to:
`    let userid_res: ResultSet = session.execute_query(userid_query).await?.try_into()?;
    let userid: Uuid;
    if let Some(row) = userid_res.rows.first() {
        userid = row.try_take(0).try_into()?;
    };`

but this doesn't work either.

> > I should note, it may be helpful to expand the documentation a bit to more explicitly call this out, as with other cases like udt attributes IntoValue/TryFromValue
> 
> The docs contain a paragraph on exactly that. Is there anything missing there?

Only that I can't get it to work based on the docs. :/

> https://docs.rs/stargate-grpc/0.3.0/stargate_grpc/#working-with-uuids
> 
> > and how to handle sharing the mutable client
> 
> Don't share. Simply create multiple clients. They are very lightweight.

One of the big advantages of Http/2 is that it allows multiple requests over the same connection, reducing connection overhead and boosting perf.  Even Http/1.1 allows connection sharing, though with some limits. This advantage is lost if you have to open a connection for every request. For this use case, we can assume a high volume of concurrent requests. Protobuf gives us a bit of an edge, but I need to share connections.

**Comment by pkolaczk** [comment]
> but this doesn't work either.

Then please provide the error message. 

> One of the big advantages of Http/2 is that it allows multiple requests over the same connection

Sure, but I didn't write you should create multiple *connections*, but just create multiple *clients*.
Cloning the clients is cheap and doesn't create more connections:

Tonic docs:
```
/// `Channel` provides a
/// `Clone` implementation that is _cheap_. This is because at the very top level
/// the channel is backed by a `tower_buffer::Buffer` which runs the connection
/// in a background task and provides a `mpsc` channel interface. Due to this
/// cloning the `Channel` type is cheap and encouraged.
```

**Comment by pkolaczk** [comment]
In just released 0.4.0, you can try:

```rust
let row = userid_res.rows.first().unwrap();
let (userid,): (Uuid,) = row.try_into()?;
```
